addons/mail/models/mail_thread.py

```diff
@@ -3167,15 +3167,17 @@ def _notify_thread_by_inbox(self, message, recipients_data, msg_vals=False, **kw
                 'res_partner_id': pid,
             } for pid in inbox_pids]
             self.env['mail.notification'].sudo().create(notif_create_values)
-
-            MailMessage = self.env['mail.message']
-            messages_format_prepared = MailMessage._message_format_personalized_prepare(
-                message._message_format(msg_vals=msg_vals, for_current_user=True), partner_ids=inbox_pids)
             for partner_id in inbox_pids:
+                # sudo: res.partner to format personalized message
+                partner = self.env['res.partner'].sudo().browse(partner_id).with_prefetch(inbox_pids)
```

Comment:
Why not just browse all ids from the getgo and just to partner.id later? seems more idiomatic